assimilation_code/modules/assimilation/algorithm_info_mod.f90

```diff
@@ -128,107 +290,198 @@ subroutine probit_dist_info(kind, is_state, is_inflation, dist_type, &
+possible_filter_kinds(3) = 'UNBOUNDED_RHF'
+possible_filter_kinds(4) = 'GAMMA_FILTER'
+possible_filter_kinds(5) = 'BOUNDED_NORMAL_RHF'
+
+possible_filter_kind_ints(1) = 1
+possible_filter_kind_ints(2) = 2
+possible_filter_kind_ints(3) = 8
+possible_filter_kind_ints(4) = 11
+possible_filter_kind_ints(5) = 101
+
+!use default values if qcf_table_filename is not in namelist
+if (.not. qcf_table_listed) then
    filter_kind = BOUNDED_NORMAL_RHF
    bounded_below = .false.;    bounded_above = .false.
    lower_bound   = missing_r8; upper_bound   = missing_r8
-elseif(obs_kind == QTY_TRACER_CONCENTRATION) then
-   filter_kind = BOUNDED_NORMAL_RHF
-   bounded_below = .true.; bounded_above = .false.
-   lower_bound   = 0.0_r8; upper_bound   = missing_r8
-elseif(obs_kind == QTY_TRACER_SOURCE) then
-   filter_kind = BOUNDED_NORMAL_RHF
-   bounded_below = .true.; bounded_above = .false.
-   lower_bound   = 0.0_r8; upper_bound   = missing_r8
-else
-   write(*, *) 'Illegal obs_kind in obs_error_info'
-   stop
+   sort_obs_inc = .false.; spread_restoration = .false.
+   return
 endif
 
-! Default settings for now for Icepack and tracer model tests
-sort_obs_inc = .false.
-spread_restoration = .false.
+!get actual name of QTY from integer index
+qty_name = get_name_for_quantity(obs_qty)
+
+!find location of QTY in qcf_table_data structure
+QTY_loc = findloc(qcf_table_row_headers, qty_name)
+
+if (QTY_loc(1) == 0) then
+   !use default values if QTY is not in table
+   filter_kind = BOUNDED_NORMAL_RHF
+   bounded_below = .false.;    bounded_above = .false.
+   lower_bound   = missing_r8; upper_bound   = missing_r8
+   sort_obs_inc = .false.; spread_restoration = .false.
+   ! Default settings for now for Icepack and tracer model tests (sort_obs_inc, spread_restoration)
+
+   else
+
+      ! Comparing the filter_kind in string format to list of potential filter_kinds
+      filter_kind_string = qcf_table_data(QTY_loc(1))%obs_inc_info%filter_kind
+      call to_upper(filter_kind_string)
+      filter_kind_loc = findloc(possible_filter_kinds, trim(filter_kind_string))
+
+      if (filter_kind_loc(1) == 0) then
+         write(errstring, *) 'Invalid filter_kind: ', trim(filter_kind_string)
+         call error_handler(E_ERR, 'obs_inc_info:', errstring, source)
+
+      else
+         filter_kind = possible_filter_kind_ints(filter_kind_loc(1))
+      endif
+     
+      sort_obs_inc = qcf_table_data(QTY_loc(1))%obs_inc_info%sort_obs_inc
+      spread_restoration = qcf_table_data(QTY_loc(1))%obs_inc_info%spread_restoration
+      bounded_below = qcf_table_data(QTY_loc(1))%obs_inc_info%bounded_below
+      bounded_above = qcf_table_data(QTY_loc(1))%obs_inc_info%bounded_above
+      lower_bound = qcf_table_data(QTY_loc(1))%obs_inc_info%lower_bound
+      upper_bound = qcf_table_data(QTY_loc(1))%obs_inc_info%upper_bound
+
+endif
 
 ! Only need to set these two for options the original RHF implementation
 !!!rectangular_quadrature = .true.
```

Comment:
Do we have enough information to set appropriate values for these:
```
! Only need to set these two for options the original RHF implementation
!!!rectangular_quadrature = .true.
!!!gaussian_likelihood_tails = .false.
```

Reply:
Not yet. I discussed this with Jeff, and he said to just make a note of it for now. I have it mentioned in the open issues section of this PR

Reply:
@jlaucar this is the second one - what should be done with rectangular_quadrature and gaussian_likelihood_tails?